### ISSUE TITLE
Epic feature request: Window should be totally customizable

### ISSUE THREAD

**Issue by BieleckiLtd** [dotnet/wpf#5854]
What we, developers need in late 2021 is to have full control over the Windows that we design without having to spend hours to fix all gotchas or use third-party libraries. What I mean by control over the Window is being able to simply customize the caption area and window borders (or lack of thereof if we wish). One could say all you need to do is to set `WindowStyle="None"` which is true, but now we have to reinvent the wheel by fixing lots of issues and re-implementing lost functionality.

1. Allow customizing appearance and content. See VS Code or Teams - application real estate is not wasted; there are menus and search fields, custom buttons, etc.
2. With `WindowStyle="None"` one has to re-create min, max/restore and close buttons and implement them. W11 snapping tooltip will not be supported anyway (I believe there is already an open issue to add this).
3. Allow to easily set window draggable area or control (where you can click and drag the window around)
4. Fix window placement when maximized. Currently, window will cover the taskbar and will ever so slightly be bigger than what we can actually fit on the screen - we have to add about 7px of padding or border when maximized to fix that.

To summarize, the amount of work we have to do for our apps to feel nice and modern is crazy and we would like to have a better experience without the need to re-invent the wheel. For anyone wanting to challenge themself here is the most-comprehensive walkthrough I could find to achieve what I described: https://engy.us/blog/2020/01/01/implementing-a-custom-window-title-bar-in-wpf/

**Comment by Symbai** [comment]
Before reinventing the wheel and spend a lot of work on something where thousand third party solutions exist I would rather see the work spend on something useful where no solution exist yet, like `trimming` or `AOT`. That's something only the WPF team can do. If they don't work on that, WPF developers will never get it. Unlike yours.

**Comment by BieleckiLtd** [comment]
I agree, workarounds exist, but the whole idea of WPF being UI framework of choice becomes difficult to justify if basic design principles are so difficult to implement.

**Comment by miloush** [comment]
> 3. Allow to easily set window draggable area or control (where you can click and drag the window around)

Btw. you can call `DragMove()` on a window/mouse down event handler, can't really be much easier.

**Comment by jogibear9988** [comment]
Wasn't WindowChrome a Solution for this? https://docs.microsoft.com/de-de/dotnet/api/system.windows.shell.windowchrome?view=windowsdesktop-6.0

**Comment by jogibear9988** [comment]
And years ago I used this: [CustomChromeLibrary](https://github.com/dotnetprojects/CustomChromeLibrary), wich I think I forked from: https://www.codeproject.com/Articles/131515/WPF-Custom-Chrome-Library but don't know for sure

**Comment by JamesBalisciano** [comment]
WindowChrome, along with just some of its many deficiencies, is discussed in the linked tutorial.
